tests/test_cli.py

```diff
@@ -56,3 +58,32 @@ def test_setup_clients(opt, dest, val):
         args = setup_clients().parse_args()
         t = type(val)
         assert t(getattr(args, dest)) == val
+
+
+class Test_analyze_data(object):
```

Comment:
Good addition.
Quick tip from my experience. I may be wrong, but I found no reason to use classes in pytest. If you use just functions and fixtures it works like charm. :coffee: talk though :wink:

Reply:
I removed the class here. However for the `Test_run_analsis` I would just keep it for better visibility :)